include/RAJA/policy/hip/atomic.hpp

```diff
@@ -474,6 +474,32 @@ RAJA_INLINE __device__ T hip_atomicCAS( T volatile *acc, T compare, T val)
  *
  * These are atomic in hip device code and non-atomic otherwise
  */
+
+RAJA_SUPPRESS_HD_WARN
+template <typename T, typename host_policy>
+RAJA_INLINE RAJA_HOST_DEVICE T
+atomicLoad(hip_atomic_explicit<host_policy>, T volatile *acc)
+{
+#if defined(__HIP_DEVICE_COMPILE__)
+  return detail::hip_atomicAdd(acc, (T)0);
```

Comment:
I'm pretty sure there is a builtin for hip atomic load and store, https://github.com/ROCm/rocPRIM/blob/4458244e6dc78e666997d7552052798d069d18aa/rocprim/include/rocprim/intrinsics/atomic.hpp#L80.

Reply:
I tried these out and they were a bit faster on some platforms, but they also aren't exactly documented so I don't know what level of support they have long term.

Reply:
I can't find any documentation for it, or when it was added. Though it looks like it wasn't used in rocprim until rocm 6. Do you have an example of conditionally doing something for a specific rocm version? Would I just need to check for a specific value of `__HIP_DEVICE_COMPILE__`?

Reply:
That would be pretty good. Maybe a check on `__has_builtin` would be useful as hip always uses clang https://releases.llvm.org/3.1/tools/clang/docs/LanguageExtensions.html.

Reply:
For checking the hip version, https://github.com/LLNL/RAJA/blob/develop/include/RAJA/config.hpp.in#L274.

Reply:
I'm now using the hip builtin atomics if they are available.